### PR TITLE
Paginate the index resource

### LINKED ISSUE #2373: Changes in `Indexes` resources
## Specifications

- Naming changes: https://github.com/meilisearch/specifications/pull/140
- Paginate the resource: https://github.com/meilisearch/specifications/pull/138

## Changes

- [ ] Remove the `name` field in every response of the `/indexes` endpoints
- [ ] Paginate the resources
  - [ ] Return the `Indexes` API resources on `GET - /indexes` in an array in a `results` field
  - [ ]Add `limit` (Default: `20`) and `offset`(Default: `0`) query parameters.
  - [ ] Add `limit`, `offset` and `total` in the response body.

Pinging @meilisearch/integration-team @meilisearch/docs-team

### PR DESCRIPTION
**Pull request by irevoire**
Fix #2373

### CODE REVIEW

File: meilisearch-http/src/routes/indexes/mod.rs [hunk 1]
```diff
@@ -39,19 +39,38 @@ pub fn configure(cfg: &mut web::ServiceConfig) {
...
         .collect();
 
     debug!("returns: {:?}", indexes);
-    Ok(HttpResponse::Ok().json(indexes))
+    Ok(HttpResponse::Ok().json(json!({
+        "results": indexes,
+        "offset": paginate.offset,
+        "limit": paginate.limit,
+        "total": nb_indexes,
+    })))
```
Comment (MarinPostma) [inline meilisearch-http/src/routes/indexes/mod.rs:73]: can we create a type for that? `IndexListView`

**Review by MarinPostma — CHANGES_REQUESTED** [review]

**Review by MarinPostma — DISMISSED** [review]
bors merge

**Comment by Kerollmops** [comment]
@MarinPostma do you have any clue about [what's happening here](https://github.com/meilisearch/meilisearch/runs/6673948825?check_suite_focus=true)?

**Comment by MarinPostma** [comment]
@irevoire is dealing with that

**Review by MarinPostma — APPROVED** [review]
bors merge

**Pull request merged**